packages/core/helper-plugin/src/features/StrapiApp.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+
+import { TranslationMessage } from 'types';
+
+interface Permission {
+  action: string;
+  subject: string | null;
+}
+
+interface MenuItem {
+  to: string;
+  icon: React.ComponentType;
+  intlLabel: TranslationMessage;
+  permissions?: Permission[];
+  Component?: React.ComponentType;
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * Context
+ * -----------------------------------------------------------------------------------------------*/
+
+interface Plugin {
+  apis: Record<string, unknown>;
+  injectionZones: Record<string, unknown>;
+  initializer: ({ setPlugin }: { setPlugin: (pluginId: string) => void }) => null | null;
+  isReady: boolean;
+  name: string;
+  pluginId: string;
+}
+
+interface StrapiAppSettingLink {
+  id: string;
+  to: string;
+  intlLabel: TranslationMessage;
+  Component: React.ComponentType;
+  permissions: Permission[] | [];
+}
+
+interface StrapiAppSetting {
+  id: string;
+  intlLabel: TranslationMessage;
+  links: StrapiAppSettingLink[];
+}
+
+interface RunHookSeries {
+  (hookName: string, asynchronous: true): Promise<unknown>;
+  (hookName: string, asynchronous: false | undefined): unknown;
+}
+
+interface RunHookWaterfall {
+  <InitialValue, Store>(
+    hookName: string,
+    initialValue: InitialValue,
+    asynchronous: false | undefined,
+    store: Store
+  ): unknown;
+  <InitialValue, Store>(
+    hookName: string,
+    initialValue: InitialValue,
+    asynchronous: true,
+    store: Store
+  ): Promise<unknown>;
+}
```

Comment:
```suggestion
type RunHookWaterfall<InitialValue, Store> = (
    hookName: string,
    initialValue: InitialValue,
    asynchronous: false | undefined,
    store: Store
  ) => unknown | Promise<unknown>;
```

imo i think this is fine :)